apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4ExplanationOfBenefitResourceProvider.java

```diff
@@ -195,6 +197,14 @@ public ExplanationOfBenefit read(@IdParam IdType eobId, RequestDetails requestDe
                 new TransformerContext(
                     metricRegistry, Optional.of(includeTaxNumbers), drugCodeDisplayLookup),
                 claimEntity);
+
+    // Add bene_id to MDC logs
+    if (eob.getPatient() != null && !Strings.isNullOrEmpty(eob.getPatient().getReference())) {
+      String beneficiaryId = eob.getPatient().getReference().replace("Patient/", "");
+      if (!Strings.isNullOrEmpty(beneficiaryId)) {
+        LoggingUtils.logBeneIdToMdc(Long.parseLong(beneficiaryId));
```

Comment:
we need to wrap this in a try/catch in case the string is malformed somehow so that it doesn't cause the request to fail.

Reply:
Refactored